src/ucs/sys/sys.c

```diff
@@ -330,25 +330,38 @@ size_t ucs_get_page_size()
     return page_size;
 }
 
-size_t ucs_get_huge_page_size()
+int ucs_get_meminfo_entry(char *pattern)
 {
-    static size_t huge_page_size = 0;
     char buf[256];
-    int size_kb;
+    char final_pattern[80];
+    int val = 0;
     FILE *f;
+    f = fopen("/proc/meminfo", "r");
+    if (f != NULL) {
+        strcpy(final_pattern, pattern);
+        strcat(final_pattern, ":      %d kB");
```

Comment:
pls use snprintf instead of strcpy+strcat
also, need just one space, according to scanf man page - a space matches a sequence of one or more spaces